@here/harp-mapview/lib/MapAnchors.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { StylePriority } from "@here/harp-datasource-protocol";
+import {
+    GeoCoordinates,
+    GeoCoordLike,
+    isGeoCoordinatesLike,
+    isVector3Like,
+    Projection,
+    Vector3Like
+} from "@here/harp-geoutils";
+
+import * as THREE from "three";
+
+/**
+ * An interface describing [[THREE.Object3D]]s anchored on given [[GeoCoordinates]].
+ *
+ * Example:
+ * ```typescript
+ * const mesh: MapAnchor<THREE.Mesh> = new THREE.Mesh(geometry, material);
+ * mesh.anchor = new GeoCoordinates(latitude, longitude, altitude);
+ * mapView.mapAnchors.add(mesh);
+ * ```
+ */
+export type MapAnchor<T extends THREE.Object3D = THREE.Object3D> = T & {
+    /**
+     * The position of this [[MapAnchor]] in [[GeoCoordinates]].
+     * @deprecated Use [[anchor]] instead.
+     */
+    geoPosition?: GeoCoordinates;
+
+    /**
+     * The anchor of this Object3D in [[GeoCoordinates]] or world coordinates.
+     */
+    anchor?: GeoCoordLike | Vector3Like;
+
+    /**
+     * Flag defining if the object may be picked.
+     *
+     * @note By default all objects are pickable even if this flag is undefined.
+     */
+    pickable?: boolean;
+
+    /**
+     * The styleSet that owns this map object.
+     *
+     * @remarks
+     * This property is used together with [[Theme.priorities]] to compute the render
+     * order of this map object.
+     */
+    styleSet?: string;
+
+    /**
+     * The category of this style.
+     *
+     * @remarks
+     * This property is used together with [[Theme.priorities]] to compute the render
+     * order of this mao object.
+     */
+    category?: string;
```

Comment:
we could use a reserved styleSet/category combination to identify the objects that need to be overlayed on top of the map scene (after text, markers and extruded buildings are rendered), alternatively we may need to add a property to MapAnchor that can be used to specify the scene/pass.

Reply:
yes, I wanted to discuss that as well. I will add the overlay scene in a separate PR though.